ls1intum/Artemis
# Feature/filter exercises

### Checklist


### Motivation and Context


We currently have no way of filtering exercises in the overview (see #386). There should at least be some options of filtering the displayed exercises.

### Description

I added a new filter dropdown next to the sorting dropdown. The students can now select between multiple options of filtering the displayed exercise list:

- **Select all**: Selects all options
- **Select none**: Selects no options
- **Full score**: Hides/displays exercises for which the students already have an assessment of 100%
- **Overdue**: Hides/displays exercises for which the due date lies in the past

### Steps for Testing


1. Log in to ArTEMiS
2. Navigate to any course
3. Click on the filter dropdown next to the sorting dropdown
4. Select any combination of filters and sorting options
5. The view should change accordingly and the exercises should be displayed with the applied filters and in the selected order

### Screenshots


![filter_exercises](https://user-images.githubusercontent.com/27979674/64171233-6f369900-ce52-11e9-9705-3ccf047018cf.gif)
